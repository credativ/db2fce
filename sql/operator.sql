-- VALUE()
SELECT VALUE(1) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR
SELECT VALUE(NULL) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR
SELECT VALUE(NULL, NULL) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR
SELECT VALUE(NULL, 1, NULL) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR
SELECT VALUE(NULL, '1', NULL) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR

-- ^= operator
SELECT 1 FROM SYSIBM.SYSDUMMY1 WHERE 1 ^= 2;
SELECT 1 FROM SYSIBM.SYSDUMMY1 WHERE 1 ^= 1;

-- !! operator
SELECT 'FOO' !! ' BAR' FROM SYSIBM.SYSDUMMY1;
