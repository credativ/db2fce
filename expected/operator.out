-- VALUE()
SELECT VALUE(1) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR
 value 
-------
     1
(1 row)

SELECT VALUE(NULL) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR
 value 
-------
 
(1 row)

SELECT VALUE(NULL, NULL) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR
 value 
-------
 
(1 row)

SELECT VALUE(NULL, 1, NULL) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR
 value 
-------
     1
(1 row)

SELECT VALUE(NULL, '1', NULL) FROM SYSIBM.SYSDUMMY1; -- DB2 ERROR
 value 
-------
 1
(1 row)

-- ^= operator
SELECT 1 FROM SYSIBM.SYSDUMMY1 WHERE 1 ^= 2;
 ?column? 
----------
        1
(1 row)

SELECT 1 FROM SYSIBM.SYSDUMMY1 WHERE 1 ^= 1;
 ?column? 
----------
(0 rows)

-- !! operator
SELECT 'FOO' !! ' BAR' FROM SYSIBM.SYSDUMMY1;
 ?column? 
----------
 FOO BAR
(1 row)

